src/sqlfluff/rules/L057.py

```diff
@@ -89,6 +89,41 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
                     identifier = identifier[:-1]
                 identifier = identifier.replace(".", "")
 
+            # Spark3 file references for direct file query
+            # are quoted in back ticks to allow for dots and regex patterns
+            # https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-file.html
+            #
+            # Path Glob Filters (done inline for SQL direct file query)
+            # https://spark.apache.org/docs/latest/sql-data-sources-generic-options.html#path-global-filter  # noqa: E501
+            #
+            # Paths also allow for "/" and "."
+
+            # Strip out special characters before testing the identifier
+            spark3_allowed_identifiers = [
+                ".",
+                "/",
+                "\\",
+                "*",
+                "?",
+                "[",
+                "]",
+                "^",
+                "-",
+                "{",
+                "}",
+                ",",
+                " ",
+            ]
+
+            if (
+                context.dialect.name in ["spark3"]
+                and context.parent_stack
+                and context.parent_stack[-1].name == "FileReferenceSegment"
+            ):
+                for allowed in spark3_allowed_identifiers:
+                    if allowed in identifier:
+                        identifier = identifier.replace(allowed, "")
```

Comment:
Seems like a lot. Rather than allowing them piece by piece I wonder if just easier to ignore these fields from this rule:

```suggestion
            # Ignore file names as are expected to contain a lot of characters
            if (
                context.parent_stack
                and context.parent_stack[-1].name == "FileReferenceSegment"
            ):
                return None
```